contrib/pkg/clusterpool/clusterpool.go

```diff
@@ -0,0 +1,419 @@
+create-pool CLUSTER_POOL_NAME --cloud=gcp`,
+		Short: "Creates a new Hive cluster pool",
+		Long:  fmt.Sprintf(longDesc, defaultPullSecretFile),
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			log.SetLevel(log.InfoLevel)
+			if err := opt.complete(args); err != nil {
+				log.WithError(err).Fatal("Error")
+			}
+			if err := opt.validate(cmd); err != nil {
+				log.WithError(err).Fatal("Error")
+			}
+			err := opt.run()
+			if err != nil {
+				log.WithError(err).Fatal("Error")
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opt.Cloud, "cloud", cloudAWS, "Cloud provider: aws(default)|azure|gcp)")
+	flags.StringVarP(&opt.Namespace, "namespace", "n", "", "Namespace to create cluster pool in")
+	flags.StringVar(&opt.BaseDomain, "base-domain", "new-installer.openshift.com", "Base domain for the cluster pool")
+	flags.StringVar(&opt.PullSecret, "pull-secret", "", "Pull secret for cluster pool. Takes precedence over pull-secret-file.")
+	flags.StringVar(&opt.PullSecretFile, "pull-secret-file", defaultPullSecretFile, "Pull secret file for cluster pool")
+	flags.StringVar(&opt.CredsFile, "creds-file", "", "Cloud credentials file (defaults vary depending on cloud)")
+	flags.StringVar(&opt.CloudSecretName, "cloud-secret", "", "Cloud secret name if already created")
+	flags.StringVar(&opt.ClusterImageSet, "image-set", "", "Cluster image set to use for this cluster pool")
+	flags.StringVar(&opt.ReleaseImage, "release-image", "", "Release image to use for installing the clusters in this pool")
+	flags.StringVar(&opt.ReleaseImageSource, "release-image-source", "https://openshift-release.svc.ci.openshift.org/api/v1/releasestream/4-stable/latest", "URL to JSON describing the release image pull spec")
+	flags.StringVar(&opt.Region, "region", "", "Region to which to install the cluster pool.")
+	flags.Int32Var(&opt.Size, "size", 1, "Size of cluster pool")
+	flags.StringVar(&opt.AzureBaseDomainResourceGroupName, "azure-base-domain-resource-group-name", "os4-common", "Resource group where the azure DNS zone for the base domain is found")
+	return cmd
+}
+
+// complete finishes parsing arguments for the command
+func (o *ClusterPoolOptions) complete(args []string) error {
+	o.Name = args[0]
+
+	if o.Region == "" {
+		switch o.Cloud {
+		case cloudAWS:
+			o.Region = "us-east-1"
+		case cloudAzure:
+			o.Region = "centralus"
+		case cloudGCP:
+			o.Region = "us-east1"
+		}
+	}
+
+	if len(o.CloudSecretName) == 0 {
+		o.createCloudSecret = true
+	}
+
+	return nil
+}
+
+// validate ensures that option values make sense
+func (o *ClusterPoolOptions) validate(cmd *cobra.Command) error {
+	if len(o.Name) == 0 {
+		return fmt.Errorf("name is required")
+	}
+	if len(o.BaseDomain) == 0 {
+		return fmt.Errorf("base domain is required")
+	}
+	if (len(o.ClusterImageSet) != 0 && len(o.ReleaseImage) != 0) ||
+		(len(o.ClusterImageSet) != 0 && len(o.ReleaseImageSource) != 0) {
```

Comment:
nit: `(len(o.ClusterImageSet) != 0 && ( len(o.ReleaseImage) != 0) ||  len(o.ReleaseImageSource) != 0))` in the if expression will give same the result?